### PR TITLE
Add support for alarm_control_panel (Fixes #19)

### LINKED ISSUE #19: Alarm support planned?
Any plans to also support Home Assistant alarms or is the idea that the goodnight scene in the Home/Homekit apps simply actives a similar scene in Home Assistant.

Not sure what other use cases there are, but the ultimate goal is to only expose Home Assistant to the Internet via Home/Homekit in order to piggyback on Apple's security between devices.

### PR DESCRIPTION
**Pull request by schmittx**
Adds support for alarm control panels, fixes #19.

Tested with Honeywell TotalConnect in Home Assistant, all seems to be working. Please feedback if you experience problems.

Due to lack of "arm_night" service in Home Assistant, selecting "Night" in HomeKit will set system to home/stay/etc. I'm currently working to submit a PR for Home Assistant to add this ability and will update this plugin when available.

### CODE REVIEW

File: accessories/alarm_control_panel.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      var alarmState = 3;
+    } else if (newState.state === 'triggered') {
+      var alarmState = 4;
+    } else {
+      var alarmState = 3;
+    }
+    this.alarmService.getCharacteristic(Characteristic.SecuritySystemCurrentState)
+        .setValue(alarmState, null, 'internal');
+    this.alarmService.getCharacteristic(Characteristic.SecuritySystemTargetState)
+        .setValue(alarmState, null, 'internal');
```
Comment (schmittx) [inline accessories/alarm_control_panel.js]: I'm new to this, any suggestions on how to fix (if needed)?

File: index.js [hunk 2]
```diff
@@ -211,6 +214,7 @@ function HomebridgeHomeAssistant(homebridge) {
   HomeAssistantBinarySensorFactory = require('./accessories/binary_sensor')(Service, Characteristic, communicationError);
   HomeAssistantDeviceTrackerFactory = require('./accessories/device_tracker')(Service, Characteristic, communicationError);
   HomeAssistantClimate = require('./accessories/climate')(Service, Characteristic, communicationError);
+  HomeAssistantAlarmControlPanel = require('./accessories/alarm_control_panel')(Service, Characteristic, communicationError);
```
Comment (schmittx) [inline index.js]: Does this need to be fixed? I'm not sure how to do...

**Comment by robbiet480** [comment]
Thanks! 🐬 🍪 💯

**Comment by JAYMAN-ATX** [comment]
Is that Temporary workaround for "NIGHT_ARM" still required? 

https://github.com/home-assistant/homebridge-homeassistant/pull/176/commits/01059c1e93ddb5ffbea3acee6dab2ff2753975e8

I can set my Manual Alarm Control Panel in Home Assistant to STATE_ALARM_ARMED_NIGHT just fine, and when I do so, it is also correctly reflected in HomeKit, but the other way doesn't work, since setting the state to NIGHT_ARM in HomeKit will be translated to 'alarm_arm_home' by Homebridge.

I see where it does that in the source code, but unfortunately I just got into Homebridge, so I have no idea how to make that change and test it.

**Pull request merged**